creator/extract_configs/views.py

```diff
@@ -0,0 +1,122 @@
+import urllib
+from typing import Optional
+from django.template.loader import render_to_string
+from django.http import HttpResponse, HttpResponseNotFound, JsonResponse
+from django.shortcuts import render
+from django.conf import settings
+from django.core.files.base import ContentFile
+from django_s3_storage.storage import S3Storage
+from botocore.exceptions import ClientError
+from creator.files.models import File, Version, DevDownloadToken, DownloadToken
+
+
+def _resolve_version(
+    file_id: str, version_id: Optional[str]
+) -> (File, Version):
+    """
+    Returns a version either specified by the version_id, or the file's latest
+    version if no version_id is specified
+    """
+    file = File.objects.get(kf_id=file_id)
+    if version_id:
+        obj = file.versions.get(kf_id=version_id)
+    else:
+        obj = file.versions.latest("created_at")
+    return file, obj
+
+
+def download_config(request, study_id, file_id, version_id=None):
+    """
+    Allow download if user is an admin OR user belongs to the file's study
+    OR if there is an unclaimed token provided that has not expired.
+    """
+    TYPE_TEMPLATE = {"S3S": "s3_scrape_config.py"}
```

Comment:
Can you put this into the FILE_TYPES in here? https://github.com/kids-first/kf-api-study-creator/blob/master/creator/analyses/file_types.py
Though we may eventually want to move that file to the settings.